### PR TITLE
ci: Use Homebrew addon on native macOS

### PR DESCRIPTION
**Pull request by hebasto**
Recently almost every macOS image update on Travis breaks our builds:
- #17848
- #18436 

This PR:
- fixes the error caused by the recent [update](https://changelog.travis-ci.com/xcode-11-3-1-xcode-11-2-1-xcode-11-1-and-xcode11-images-updated-142286) from 10.14.4 (18E226) to 10.14.6 (18G3020) on March 25
- leverages [Homebrew addon](https://config.travis-ci.com/ref/job/addons/homebrew) to install packages

Homebrew is not told to install `automake` and `pkg-config` packages, as the [docs](https://docs.travis-ci.com/user/reference/osx/#compilers-and-build-toolchain) states that they are pre-installed:
>    - automake 1.16.1
>    - pkg-config 0.29.2

### CODE REVIEW

File: ci/test/04_install.sh [hunk 1]
```diff
@@ -19,18 +19,6 @@ if [ "$TRAVIS_OS_NAME" == "osx" ]; then
   git reset --hard origin/master
```
Comment (MarcoFalke) [inline ci/test/04_install.sh]: Why is this needed?

Comment (hebasto) [reply ci/test/04_install.sh]: No need. Removed.

**Comment by hebasto** [comment]
Updated ae9b1e2faa960293345513c0b86666c26a113730 -> 25c8b73656c90f7b78a90f25c5bcc308f8b7c598 ([pr18438.01](https://github.com/hebasto/bitcoin/commits/pr18438.01) -> [pr18438.02](https://github.com/hebasto/bitcoin/commits/pr18438.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18438.01..pr18438.02)):

- @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18438#discussion_r398541716) addressed

**Comment by fanquake** [comment]
Concept ACK. Thanks for following up with this.

**Comment by MarcoFalke** [comment]
Seems to pick up everything correctly (wallet, ccache, qt):

```
Options used to compile and link:

  with wallet   = yes

  with gui / qt = yes

    with qr     = yes

  with zmq      = yes

  with test     = yes

    with prop   = no

    with fuzz   = no

  with bench    = yes

  with upnp     = yes

  use asm       = yes

  sanitizers    = 

  debug enabled = no

  gprof enabled = no

  werror        = yes

  target os     = darwin

  build os      = darwin

  CC            = /usr/local/bin/ccache gcc

  CFLAGS        = -g -O2

  CPPFLAGS      =   -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Qunused-arguments  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -I/usr/local/opt/berkeley-db@4/include -DMAC_OSX -DOBJC_OLD_DISPATCH_PROTOTYPES=0

  CXX           = /usr/local/bin/ccache g++ -std=c++11

  CXXFLAGS      =   -Wstack-protector -fstack-protector-all  -Wall -Wextra -Wformat -Wvla -Wswitch -Wformat-security -Wthread-safety-analysis -Wrange-loop-analysis -Wredundant-decls -Wunused-variable -Wdate-time  -Wno-unused-parameter -Wno-self-assign -Wno-unused-local-typedef -Wno-deprecated-register -Wno-implicit-fallthrough  -Werror=vla -Werror=switch -Werror=thread-safety-analysis -Werror=unused-variable -Werror=date-time -Werror=return-type  -g -O2 -fvisibility=hidden

  LDFLAGS       = -pthread  -Wl,-bind_at_load   -Wl,-headerpad_max_install_names -Wl,-dead_strip -Wl,-dead_strip_dylibs

  ARFLAGS       = cr
